crates/primitives/src/chain/spec.rs

```diff
@@ -573,6 +573,14 @@ impl ChainSpec {
 
     /// Returns `true` if this chain contains Optimism configuration.
     #[inline]
+    #[cfg(feature = "optimism")]
```

Comment:
Note: This change attempts to preserve the `const` behavior when possible, and also prioritizes the `const` `self.chain.is_optimism()` case over the slower runtime-evaluated hardfork check. The performance penalty should only impact unnamed chains or non-OP chains when the optimism feature is enabled.

Reply:
nice